frontend/src/metabase/dashboard/actions/cards.js

```diff
@@ -71,6 +76,33 @@ export const addCardToDashboard =
     dispatch(loadMetadataForDashboard([dashcard]));
   };
 
+export const removeCardFromDashboard = createThunkAction(
+  REMOVE_CARD_FROM_DASH,
+  ({ dashcardId, cardId }) =>
+    (dispatch, _getState) => {
+      dispatch(cancelFetchCardData(cardId, dashcardId));
+      return { dashcardId, cardId };
+    },
+);
+
+export const undoRemoveCardFromDashboard = createThunkAction(
+  UNDO_REMOVE_CARD_FROM_DASH,
+  ({ dashcardId }) =>
+    (dispatch, getState) => {
+      const dashcard = getDashCardById(getState(), dashcardId);
+      const card = dashcard.card;
+
+      const cardAlreadyLoaded = Boolean(
+        getSingleDashCardData(getState(), dashcardId),
+      );
+
+      if (!cardAlreadyLoaded) {
+        dispatch(fetchCardData(card, dashcard));
```

Comment:
It seems that `fetchCardData` already checks for loaded data, so these checks might be not needed